controllers/articles.js

```diff
@@ -0,0 +1,195 @@
+   */
+  deleteArticle: async (req, res) => {
+    const { slug } = req.params;
+    const row = await article.update({ deleted: 1 }, { where: { slug, deleted: 0 } });
+    if (row[0] === 0) {
+      return res.status(404).send({
+        status: res.statusCode,
+        message: 'No article found for this slug',
+      });
+    }
+    return res.status(200).send({
+      status: res.statusCode,
+      message: 'Article deleted successfully',
+    });
+  },
+
+  /*
+  * Retieving all the published articles based to the author
+  * and the status of the article (Published).
+  */
+  getAllPublished: async (req, res) => {
+    try {
+      // const { author } = req.user;
+      const response = await article.findAll({ where: { author: tempUser, status: 'Published', deleted: 0 } });
+      return res.status(200).send({
+        status: res.statusCode,
+        articles: response,
+      });
+    } catch (error) {
+      throw error;
+    }
+  },
+
+  /*
+  * Retieving all the published articles based to the author
+  * and the status of the article (Draft).
+  */
+  getAllDraft: async (req, res) => {
+    // const { author } = req.user;
+    try {
+      const response = await article.findAll({ where: { author: tempUser, status: 'Draft', deleted: 0 } });
+      return res.status(200).send({
+        status: res.statusCode,
+        articles: response,
+      });
+    } catch (error) {
+      throw error;
+    }
+  },
+
+  /*
+   * Construct user's feed
+   * and the status of the article (Published).
+   */
+  getFeeds: async (req, res) => {
+    try {
+      const response = await article.findAll({
+        where: {
+          status: 'Published',
+          deleted: 0
+        }
+      });
+      return res.status(200).send({
+        status: res.statusCode,
+        articles: response,
+      });
+    } catch (error) {
+      throw error;
+    }
+  },
+
+  viewAnArticle: async (req, res) => {
+    const { slug } = req.params;
+    try {
+      const findAuthor = await User.findOne();
+      const viewOneArticle = await article.findOne({ where: { slug } });
+      if (!viewOneArticle && !findAuthor) {
+        return res.status(404).send({
+          status: res.statusCode,
+          errorMessage: 'No article found, please create an article first'
+        });
+      }
+      return res.status(200).send({
+        status: res.statusCode,
+        article: {
+          title: viewOneArticle.title,
+          body: viewOneArticle.body,
+          description: viewOneArticle.description,
+          slug: viewOneArticle.slug,
+          author: {
+            username: findAuthor.username,
+            bio: findAuthor.bio,
+            image: findAuthor.image,
+            following: findAuthor.following
+          }
+        }
+      });
+    } catch (error) {
+      if (error.message) {
+        return res.status(500).send({
+          status: res.statusCode,
+          errorMessage: error.message,
+        });
+      }
+    }
+  },
+};
+
```

Comment:
I think you should add another controller for getting a given number of articles or find a way to add it in the above controllers. This would be useful when rendering an article page as on the bottom we have to show other related articles someone can read next.
Check at the bottom of this [Medium article](https://onezero.medium.com/machine-learning-might-render-the-human-quest-for-knowledge-pointless-5425f8b00a45)

Reply:
@avpaul on line 112-123 @niyobobo is all published articles like this:
  getAllPublished: async (req, res) => {
    try {
      // const { author } = req.user;
      const response = await article.findAll({ where: { author: tempUser, status: 'Published', deleted: 0 } });
      return res.status(200).send({
        status: res.statusCode,
        articles: response,
      });
    } catch (error) {
      throw error;
    }
  },

This can also be used to get random articles that can feature in the bottom as your example above.